### PR TITLE
initial version of Champions of Meridian 

### PR DESCRIPTION
**Pull request by andygeiss**

### CODE REVIEW

**Comment by skittles1** [comment]
Just some quick notes before reviewing this fully:
- Please adjust the kod/lkod files to match the coding style of the repo:
   - Indent everything in the `messages:` section 3 spaces (lkod doesn't need this)
   - Break up lines longer than 80 chars where possible, applies to Send calls & strings mostly. Check some other kod/lkod files to see how you can split up strings in the `resources:` section over multiple lines.
- The `for` loop in `CheckChampionsForRewardRequest` looks like it should be a list iteration using `foreach` instead. I think there are other instances of for loops in this PR that need to be changed also.
- The MsgSendUser call in `CheckChampionsForRewardRequest` has typos in the parameter names (should be `parm1` and `parm2` vs `param1` and `param2`. I don't know if you've tried to compile or test this code yet but I'd prefer if you do both before submitting a PR here, or state in the PR description that it's pre-compilation so not ready for review if you want to show a concept which might be the intention here.

**Comment by andygeiss** [comment]
I adjusted the lines to match the coding style and fixed the typos.
Version compiled successfully.
The for loop is needed because the index i specifies the data of a specific hero in plChampions, plPoints, plRooms and plTimes.

**Comment by skittles1** [comment]
> specific hero in plChampions, plPoints, plRooms and plTimes.

You would need to combine this in a list of `[ [ champion, points, room, time ] , [ champion, points, room, time ], ... ]`. Performance is fairly critical in kod because it is a single-threaded interpreted language, and in this case repeatedly calling Nth on multiple different lists is inefficient compared to indexing into one list with the opcodes used underneath the hood of the `foreach` loop.

**Comment by andygeiss** [comment]
I combined the lists and replaced the current for loops with foreach.

File: kod/util/championsmaint.kod [hunk 1]
```diff
@@ -0,0 +1,881 @@
...
+   oItem = Send(self,@GetRewardItem,#who=who);
+   if oItem <> $
+   {
+      sInscription = Send(oItem,@GetInscription);
+      oClass = Send(self,@GetRewardItemClass,#sInscription=sInscription);
+      iTranslation = Send(self,@GetRewardItemTranslation,#sInscription=sInscription);
+      oTarget = Send(self,@GetRewardTarget,#who=who,#class=oClass);
+      if oTarget <> $
+      {
+         // Bug? item looks like enchanted (orange) but description will not appear on click
```
Comment (skittles1) [inline kod/util/championsmaint.kod]: The IA_BONDED attribute is not what players know of as 'bonded' or SWL. What you want here is IA_TRANSCENDANT which just needs `oItem` specified and not `oPlayer` and is the attribute used in the Bond spell/dm item bonding.

Comment (andygeiss) [reply kod/util/championsmaint.kod]: great I didnt know how to solve that! I will change that

**Comment by skittles1** [comment]
Don't forget to indent the `messages` block in championsmaint.kod by 3 spaces to match other files. Also, our convention is to use `First(list)` rather than `Nth(list,1)`, eventually I'll have the compiler modify these calls during compilation but for now it's best to write it using `First`.

**Pull request closed without merging**